runtime/drivers/athena/olap.go

```diff
@@ -0,0 +1,418 @@
+		return false
+	}
+	if r.scanErr != nil {
+		return false
+	}
+
+	// see if we have more rows in current page
+	if r.currentRow < len(r.results.ResultSet.Rows) {
+		row := r.results.ResultSet.Rows[r.currentRow]
+		var err error
+		for i, col := range row.Data {
+			if col.VarCharValue == nil {
+				r.scannedRow[i] = nil
+				continue
+			}
+			r.scannedRow[i], err = convertValue(*r.columnInfo[i].Type, *col.VarCharValue)
+			if err != nil {
+				r.scanErr = err
+				return false
+			}
+		}
+		r.currentRow++
+		return true
+	}
+	// Fetch next page if available
+	if r.results.NextToken == nil {
+		return false
+	}
+	nextResults, err := r.client.GetQueryResults(context.Background(), &athena.GetQueryResultsInput{
+		QueryExecutionId: aws.String(r.queryID),
+		NextToken:        r.results.NextToken,
+	})
+	if err != nil {
+		r.scanErr = fmt.Errorf("failed to get next page of query results: %w", err)
+		return false
+	}
+	r.results = nextResults
+	r.currentRow = 0 // there is no header in next page
+	// call Next to process the newly fetched page
+	return r.Next()
+}
+
+// Scan implements drivers.Rows.
+func (r *rows) Scan(dest ...any) error {
+	if r.scanErr != nil {
+		return r.scanErr
+	}
+	if r.scannedRow == nil {
+		return fmt.Errorf("must call Next before Scan")
+	}
+	if len(dest) != len(r.scannedRow) {
+		return fmt.Errorf("expected %d destination arguments in Scan, got %d", len(r.scannedRow), len(dest))
+	}
+
+	for i := range dest {
+		err := sqlconvert.ConvertAssign(dest[i], r.scannedRow[i])
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *rows) runtimeSchema() (*runtimev1.StructType, error) {
+	fields := make([]*runtimev1.StructType_Field, len(r.columnInfo))
+	for i, col := range r.columnInfo {
+		fields[i] = &runtimev1.StructType_Field{
+			Name: *col.Name,
+		}
+		colType, err := athenaTypeToRuntimeType(*col.Type)
+		if err != nil {
+			return nil, err
+		}
+		fields[i].Type = colType
+	}
+	res := &runtimev1.StructType{
+		Fields: fields,
+	}
+	return res, nil
+}
+
+func athenaTypeToRuntimeType(colType string) (*runtimev1.Type, error) {
+	t := &runtimev1.Type{}
+	switch colType {
+	case "tinyint":
+		t.Code = runtimev1.Type_CODE_INT8
+	case "smallint":
+		t.Code = runtimev1.Type_CODE_INT16
+	case "integer":
+		t.Code = runtimev1.Type_CODE_INT32
+	case "bigint":
+		t.Code = runtimev1.Type_CODE_INT64
+	case "float", "real":
+		t.Code = runtimev1.Type_CODE_FLOAT32
+	case "double":
+		t.Code = runtimev1.Type_CODE_FLOAT64
+	case "boolean":
+		t.Code = runtimev1.Type_CODE_BOOL
+	case "date", "time", "time with time zone", "timestamp", "timestamp with time zone":
+		t.Code = runtimev1.Type_CODE_TIMESTAMP
+	case "json", "char", "varchar", "varbinary", "row", "string", "binary",
+		"struct", "interval year to month", "interval day to second", "decimal",
+		"ipaddress", "array", "map", "unknown":
+		t.Code = runtimev1.Type_CODE_STRING
+	default:
+		return nil, fmt.Errorf("unknown type %q", colType)
```

Comment:
Is it important we get the right type here? Or would it be safer to return `runtimev1.Type_CODE_UNSPECIFIED` instead?